src/cascadia/TerminalControl/ControlInteractivity.h

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+// Module Name:
+// - ControlInteractivity.h
+//
+// Abstract:
+// - This is a wrapper for the `ControlCore`. It holds the logic for things like
+//   double-click, right click copy/paste, selection, etc. This is intended to
+//   be a UI framework-independent abstraction. The methods this layer exposes
+//   can be called the same from both the WinUI `TermControl` and the WPF
+//   control.
+//
+// Author:
+// - Mike Griese (zadjii-msft) 01-Apr-2021
+
+#pragma once
+
+#include "ControlInteractivity.g.h"
+#include "EventArgs.h"
+#include "../buffer/out/search.h"
+#include "cppwinrt_utils.h"
+
+#include "ControlCore.h"
+
+namespace Microsoft::Console::VirtualTerminal
+{
+    struct MouseButtonState;
+}
+
+namespace ControlUnitTests
+{
+    class ControlCoreTests;
+    class ControlInteractivityTests;
+};
+
+namespace winrt::Microsoft::Terminal::Control::implementation
+{
+    struct ControlInteractivity : ControlInteractivityT<ControlInteractivity>
+    {
+    public:
+        ControlInteractivity(IControlSettings settings,
+                             TerminalConnection::ITerminalConnection connection);
+
+        void GainFocus();
+        void UpdateSettings();
+        void Initialize();
+        winrt::com_ptr<ControlCore> GetCore();
+
+#pragma region InputMethods
+        void PointerPressed(const winrt::Windows::Foundation::Point mouseCursorPosition,
+                            ::Microsoft::Console::VirtualTerminal::TerminalInput::MouseButtonState buttonState,
+                            const unsigned int pointerUpdateKind,
+                            const uint64_t timestamp,
+                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                            const bool focused,
+                            const til::point terminalPosition);

```

Comment:
wait it takes both the raw _and_ cell position?

![image](https://user-images.githubusercontent.com/189190/115457876-a5bd5500-a1ea-11eb-868b-9e2976124bbf.png)